### ISSUE TITLE
`Tag` breaking in Django 4.1 when registering without the `name` kwarg

### ISSUE THREAD

**Issue by kevinmarsh** [django-cms/django-classy-tags#84]
I've just run into an issue upgrading to Django 4.1, when using `classytags.core.Tag` Django is complaining that my tag hasn't been registered. I've tracked it down to this bit of the code: https://github.com/django-cms/django-classy-tags/blob/e7eee392300e66c6f9a55d008f7fd849d6fbf8c4/classytags/core.py#L108-L124

Where it's making some assumptions about how Django's template system gets the name attribute from `_decorated_function`, however this changed under the hood in this commit in Django https://github.com/django/django/commit/3d7ac6420c0ff23305bc50c69c419c4ed5d2a838#diff-4ea91e90ebf87795fbf00d2f3bd8bb6333ef908db1adc0bc2b19f6dd5b71490f https://github.com/django/django/pull/15291

![2022-09-08_12-34](https://user-images.githubusercontent.com/1423728/189212414-fa3e1dbd-b442-4d8c-9557-f0ab1f8eca08.png)

#### Example

```python
from django import template
from classytags.core import Tag

register = template.Library()

@register.tag
class MyCustomTag(Tag):
  pass

@register.tag(name="foo_bar")
class MyCustomTagWithName(Tag):
  pass

print(register.tags)
```

In Django 4.0 (and presumably earlier) this prints 
```python
{'my_custom_tag': <class '__main__.MyCustomTag'>, 'foo_bar': <class '__main__.MyCustomTagWithName'>}
```
and in Django 4.1 this prints
```python
{'MyCustomTag': <class '__main__.MyCustomTag'>, 'foo_bar': <class '__main__.MyCustomTagWithName'>}
```

**Comment by marksweb** [comment]
@kevinmarsh Thanks for raising. I've not made it to 4.0 yet so a long way from catching this one.

If I've understood this correctly, we could do something to check if django<4.1 and vary the attribute set;

```
     def __new__(cls, name, bases, attrs): 
         parents = [base for base in bases if isinstance(base, TagMeta)] 
         if not parents: 
             return super().__new__(cls, name, bases, attrs) 
         tag_name = str(attrs.get('name', get_default_name(name))) 
  
         if django_lt_41:

             def fake_func(): 
                 pass  # pragma: no cover 
  
             fake_func.__name__ = tag_name 
             attrs['_decorated_function'] = fake_func 

        else:
            attrs['__name__'] = tag_name
```

**Comment by kevinmarsh** [comment]
Just getting back to this, think actually we might not even need `if django_lt_41`, might just be able to modify the name param for the metaclass
```diff
- return super().__new__(cls, name, bases, attrs)
+ return super().__new__(cls, tag_name, bases, attrs)
```
If you merge #85 I'll push a PR on top of that which passes on Django 2.2-4.1 (and main branch) and can discuss implementation more there

**Comment by marksweb** [comment]
@kevinmarsh I've merged #85 (sorry for the delay!)

In terms of moving forward with this, I'm more than happy to drop Django 2.2 as its no longer supported.

**Comment by fsbraun** [comment]
This issue will solve https://github.com/django-cms/django-cms/issues/7398
